src/pages/settings/ClusteredSettingFormInput.tsx

```diff
@@ -0,0 +1,93 @@
+import { FC, useState } from "react";
+import { Button, Form, Icon } from "@canonical/react-components";
+import type { ConfigField } from "types/config";
+import { getConfigId } from "./SettingForm";
+import ConfigFieldDescription from "pages/settings/ConfigFieldDescription";
+import ClusterSpecificInput from "components/forms/ClusterSpecificInput";
+import { useClusterMembers } from "context/useClusterMembers";
+import { ClusterSpecificValues } from "components/ClusterSpecificSelect";
+
+interface Props {
+  initialValue: ClusterSpecificValues;
+  configField: ConfigField;
+  onSubmit: (newValue: ClusterSpecificValues) => void;
+  onCancel: () => void;
+  readonly?: boolean;
+  toggleReadOnly?: () => void;
+}
+
+const ClusteredSettingFormInput: FC<Props> = ({
+  initialValue,
+  configField,
+  onSubmit,
+  onCancel,
+  readonly = false,
+  toggleReadOnly,
+}) => {
+  const [value, setValue] = useState<ClusterSpecificValues>(initialValue);
+
+  const { data: clusterMembers = [] } = useClusterMembers();
+  const memberNames = clusterMembers.map((member) => member.server_name);
+
+  const canBeReset = Object.values(value).some(
+    (v) => v !== configField.default,
+  );
+
+  const resetToDefault = (memberNames: string[]) => {
+    const defaultValues: { [key: string]: string } = {};
+    memberNames.forEach((name) => {
+      defaultValues[name] = configField.default;
+    });
+    setValue(defaultValues);
+  };
+
+  return (
+    <Form
+      onSubmit={(e) => {
+        e.preventDefault();
+        onSubmit(value);
+      }}
+    >
+      <ClusterSpecificInput
+        aria-label={configField.key}
+        classname="input-wrapper"
+        id={getConfigId(configField.key)}
+        values={value}
+        isReadOnly={readonly}
+        onChange={(value) => setValue(value)}
+        memberNames={memberNames}
+        toggleReadOnly={toggleReadOnly}
+        helpText={
+          <ConfigFieldDescription
+            description={configField.longdesc}
+            className="p-form-help-text"
+          />
+        }
+      />
+      {!readonly && (
+        <>
+          <Button appearance="base" onClick={onCancel}>
+            Cancel
+          </Button>
+          <Button appearance="positive" type="submit">
+            Save
+          </Button>
+          {canBeReset && (
+            <Button
+              className="reset-button"
+              appearance="base"
+              type="button"
+              onClick={() => resetToDefault(memberNames)}
```

Comment:
```suggestion
              onClick={resetToDefault}
```
We don't need the argument and can use the variable in the parent scope directly from within the function.